### PR TITLE
Adds efficientnet2 presets

### PR DESCRIPTION
**Pull request by pkgoogle**
Adds these variants:
```
efficientnet2_rw_m_agc_imagenet
efficientnet2_rw_s_ra2_imagenet
efficientnet2_rw_t_ra2_imagenet
```

Branched from Edge Presets: https://github.com/keras-team/keras-hub/pull/1976

Can just merge this one and close that one or merge that one first for more modularity.

### CODE REVIEW

File: keras_hub/src/models/efficientnet/convbnact.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        input_filters,
+        output_filters,
+        kernel_size=3,
+        strides=1,
+        data_format="channels_last",
+        batch_norm_momentum=0.9,
+        batch_norm_epsilon=1e-3,
+        activation="swish",
+        dropout=0.2,
+        nores=False,
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/efficientnet/convbnact.py:53]: add missing args in docstring - example : nores

File: keras_hub/src/models/efficientnet/convbnact.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import keras
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/efficientnet/convbnact.py:1]: rename file to be more readable - convolution_batch_norm_activation.py

File: keras_hub/src/models/efficientnet/convbnact_test.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import keras
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/efficientnet/convbnact_test.py:1]: rename file to be more readable

File: keras_hub/src/models/efficientnet/fusedmbconv.py [hunk 4]
```diff
@@ -77,6 +77,7 @@ def __init__(
         activation="swish",
         dropout=0.2,
         nores=False,
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/efficientnet/fusedmbconv.py:79]: the docstring args are not matching the arg list here - please make sure they are consistent

**Review by divyashreepathihalli — CHANGES_REQUESTED** [review]
Thanks Piseth! Left a few comments

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py [hunk 5]
```diff
@@ -170,12 +172,21 @@ def __init__(
         self._pyramid_outputs = {}
         curr_pyramid_level = 1
 
-        for i in range(len(stackwise_kernel_sizes)):
+        num_stacks = len(stackwise_kernel_sizes)
+
+        if isinstance(depth_coefficient, tuple):
```
Comment (mattdangerw) [inline keras_hub/src/models/efficientnet/efficientnet_backbone.py:177]: This is confusing, most that can be passed "stackwise" have `stackwise_` as a prefix, except `depth_coefficient` now.

It's probably better UX to just rename these first arguments `stackwise_depth_coefficient` and `stackwise_width_coefficient`. At least then our argument names are consistent. Remember to update docstrings.

You could keep backwards compat by adding a few lines at the top of the constructor here. Something like this. You have to allow `stackwise_depth_coefficient=None` and `stackwise_width_coefficient=None` in the arg list.

```python
num_stacks = len(stackwise_kernel_sizes)
if "depth_coefficient" in kwargs:
    stackwise_depth_coefficient = kwargs.pop("depth_coefficient") * num_stacks
if "width_coefficient" in kwargs:
    stackwise_width_coefficient = kwargs.pop("width_coefficient") * num_stacks
```

File: keras_hub/src/utils/timm/convert_efficientnet.py [hunk 6]
```diff
@@ -189,15 +236,22 @@ def port_batch_normalization(keras_layer, hf_weight_prefix):
 
     # Stages
     num_stacks = len(backbone.stackwise_kernel_sizes)
+
+    depth_coefficient = VARIANT_MAP[variant]["depth_coefficient"]
```
Comment (mattdangerw) [inline keras_hub/src/utils/timm/convert_efficientnet.py:240]: if we follow the suggestion about arg renaming above, we can get rid of this whole block. Just update above from

```
"width_coefficient": 0.8,
"depth_coefficient": 0.9,
```

to

```
"stackwise_width_coefficient": [0.8] * 6,
"stackwise_depth_coefficient": [0.9] * 6,
```

**Review by mattdangerw — COMMENTED** [review]
Having a partially "stackwise" arg that isn't name `"stackwise_"` is confusing. Left a suggestion to be more consistent in our arg names here. Less if branching.

File: keras_hub/src/models/efficientnet/convbnact.py (hunk 1 above)
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/convbnact.py]: done

File: keras_hub/src/models/efficientnet/convbnact.py (hunk 2 above)
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/convbnact.py]: this conflicts w/ the below review, so I chose the later review.

File: keras_hub/src/models/efficientnet/convbnact_test.py (hunk 3 above)
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/convbnact_test.py]: this conflicts w/ the below review, so I chose the later review.

File: keras_hub/src/models/efficientnet/fusedmbconv.py (hunk 4 above)
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/fusedmbconv.py:85]: done

File: keras_hub/src/models/efficientnet/efficientnet_presets.py [hunk 7]
```diff
@@ -161,6 +161,45 @@
             "path": "efficientnet",
             "model_card": "https://arxiv.org/abs/1905.11946",
         },
-        "kaggle_handle": "kaggle://keras/efficientnet/keras/efficientnet_es_ra_imagenet/1",
+        "kaggle_handle": "kaggle://keras/efficientnet/keras/efficientnet_b1_ft_imagenet",
+    },
+    "efficientnet2_rw_m_agc_imagenet": {
+        "metadata": {
+            "description": (
+                "EfficientNet-v2 Medium model trained on the ImageNet 1k "
```
[the remark this answers is not in the document]
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/efficientnet_presets.py:169]: Should we change them all? or perhaps in a different PR?

File: keras_hub/src/models/efficientnet/convbnact.py [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    "class_name": "VarianceScaling",
+    "config": {
+        "scale": 2.0,
+        "mode": "fan_out",
+        "distribution": "truncated_normal",
+    },
+}
+
+
+class ConvBNActBlock(keras.layers.Layer):
```
[the remark this answers is not in the document]
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/convbnact.py]: done

File: keras_hub/src/models/efficientnet/convbnact.py [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+            and self.input_filters == self.output_filters
+            and not self.nores
+        ):
+            if self.dropout:
+                x = self.dropout_layer(x)
+            x = keras.layers.Add(name=self.name + "add")([x, inputs])
+        return x
+
+    def get_config(self):
+        config = {
```
[the remark this answers is not in the document]
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/convbnact.py]: done

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py [hunk 10]
```diff
@@ -260,11 +271,26 @@ def __init__(
                         name=block_name,
                     )
                     x = block(x)
+                else:  # cba block
```
[the remark this answers is not in the document]
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/efficientnet_backbone.py]: Changing my mind mid-change :smile:. Refactored to make block_kwargs more dynamic now.

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py (hunk 5 above)
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/efficientnet_backbone.py]: done

File: keras_hub/src/utils/timm/convert_efficientnet.py (hunk 6 above)
Comment (pkgoogle) [reply keras_hub/src/utils/timm/convert_efficientnet.py]: I made it so both ways should work.

**Comment by pkgoogle** [comment]
I should note I'm noticing a different prediction for one variant:

```sh
python tools/checkpoint_conversion/convert_efficientnet_checkpoints.py --preset efficientnet2_rw_s_ra2_imagenet
✅ Loaded TIMM model.
I1121 13:45:18.860764 8329934912 _builder.py:196] Loading pretrained weights from Hugging Face hub (timm/efficientnetv2_rw_s.ra2_in1k)
I1121 13:45:19.120539 8329934912 _hub.py:184] [timm/efficientnetv2_rw_s.ra2_in1k] Safe alternative available for 'pytorch_model.bin' (as 'model.safetensors'). Loading weights using safetensors.
✅ Loaded KerasHub model.
1/1 ━━━━━━━━━━━━━━━━━━━━ 2s 2s/step
🔶 Keras output: [-0.575618   -0.4144789  -0.73163635 -0.8395867  -0.5637497   0.87436163
 -1.3687949  -0.1337489  -0.0662252   0.01332632]
🔶 TIMM output: [-0.7259099   0.3929606  -0.05580516  0.11954936  0.02080455  0.996287
  0.3669875   0.5366461   0.01827425 -0.37024903]
🔶 Keras label: 349
🔶 TIMM label: 345
🔶 Modeling difference: 0.52596486
🔶 Preprocessing difference: 0.0044535464
🏁 Preset saved to ./efficientnet2_rw_s_ra2_imagenet
```

Which I'm looking into

File: keras_hub/src/models/efficientnet/efficientnet_presets.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (mattdangerw) [reply keras_hub/src/models/efficientnet/efficientnet_presets.py:169]: Let's keep what we have for now! Maybe we can fix in a follow up, pending a naming discussion.

**Comment by mattdangerw** [comment]
@pkgoogle thanks! Yeah a different label and that level of different output floats worth looking at!

**Comment by pkgoogle** [comment]
Should be good now:

```sh
python tools/checkpoint_conversion/convert_efficientnet_checkpoints.py     --preset efficientnet2_rw_s_ra2_imagenet
✅ Loaded TIMM model.
I1125 14:46:30.043703 8329934912 _builder.py:196] Loading pretrained weights from Hugging Face hub (timm/efficientnetv2_rw_s.ra2_in1k)
I1125 14:46:30.252438 8329934912 _hub.py:184] [timm/efficientnetv2_rw_s.ra2_in1k] Safe alternative available for 'pytorch_model.bin' (as 'model.safetensors'). Loading weights using safetensors.
✅ Loaded KerasHub model.
1/1 ━━━━━━━━━━━━━━━━━━━━ 1s 1s/step
🔶 Keras output: [-0.37625     0.6832866   0.28495154  0.5878901   0.20605645  1.0627874
  0.40687525  0.40137056 -0.118939   -0.30322495]
🔶 TIMM output: [-0.7259099   0.3929606  -0.05580516  0.11954936  0.02080455  0.996287
  0.3669875   0.5366461   0.01827425 -0.37024903]
🔶 Keras label: 345
🔶 TIMM label: 345
🔶 Modeling difference: 0.15455398
🔶 Preprocessing difference: 0.0044535464
🏁 Preset saved to ./efficientnet2_rw_s_ra2_imagenet
```

File: keras_hub/src/models/efficientnet/cba.py [hunk 11]
```diff
@@ -0,0 +1,139 @@
+import keras
+
+BN_AXIS = 3
+
+CONV_KERNEL_INITIALIZER = {
```
Comment (mattdangerw) [inline keras_hub/src/models/efficientnet/cba.py:5]: Kinda funky to return this as a dict, any particular reason for it?

More in line with other models would be to just add a function here...

```python
def conv_kernel_initializer(scale=2.):
    return keras.initializers.VarianceScaling(
        scale=scale,
        ...
    )
```

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py [hunk 12]
```diff
@@ -99,8 +106,8 @@ class EfficientNetBackbone(FeaturePyramidBackbone):
     def __init__(
         self,
         *,
-        width_coefficient,
-        depth_coefficient,
+        stackwise_width_coefficient=None,
```
Comment (mattdangerw) [inline keras_hub/src/models/efficientnet/efficientnet_backbone.py:109]: probably make these plural, in keeping with other args? stackwise_width_coefficients and stackwise_depth_coefficients

File: keras_hub/src/utils/timm/convert_efficientnet.py [hunk 13]
```diff
@@ -91,6 +91,53 @@
...
+        "stackwise_strides": [1, 2, 2, 2, 1, 2],
+        "stackwise_squeeze_and_excite_ratios": [0, 0, 0, 0.25, 0.25, 0.25],
+        "stackwise_block_types": ["fused"] * 3 + ["unfused"] * 3,
+        "stackwise_force_input_filters": [0, 0, 0, 0, 0, 0],
+        "stackwise_nores_option": [False] * 6,
+        "activation": "silu",
+        "num_features": 1792,
+    },
+    "rw_s": {
+        "width_coefficient": 1.0,
```
Comment (mattdangerw) [inline keras_hub/src/utils/timm/convert_efficientnet.py:111]: I'd pass these the new way instead of the legacy and now undocumented path. `"stackwise...": [1.0] * 6`

File: keras_hub/src/utils/timm/convert_efficientnet.py [hunk 14]
```diff
@@ -91,6 +91,53 @@
...
+        "depth_coefficient": 1.0,
```
Comment (mattdangerw) [inline keras_hub/src/utils/timm/convert_efficientnet.py:112]: same here, and elsewhere in this file

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py [hunk 15]
```diff
@@ -26,15 +27,12 @@ class EfficientNetBackbone(FeaturePyramidBackbone):
...
-        width_coefficient: float, scaling coefficient for network width.
-        depth_coefficient: float, scaling coefficient for network depth.
-        dropout: float, dropout rate at skip connections. The default
-            value is set to 0.2.
-        depth_divisor: integer, a unit of network width. The default value is
-            set to 8.
-        activation: activation function to use between each convolutional layer.
-        input_shape: optional shape tuple, it should have exactly 3 input
-            channels.
+        stackwise_width_coefficient: list[float] or float, scaling coefficient
```
Comment (mattdangerw) [inline keras_hub/src/models/efficientnet/efficientnet_backbone.py:30]: doesn't this need to be a `list[float]`?

**Review by mattdangerw — COMMENTED** [review]
Looking good! Just little comments.

File: keras_hub/src/models/efficientnet/cba.py (hunk 11 above)
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/cba.py]: No particular reason, I was following the original style of the fusedmbconv/mbconv blocks. Done.

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py (hunk 12 above)
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/efficientnet_backbone.py]: done

File: keras_hub/src/utils/timm/convert_efficientnet.py (hunk 13 above)
Comment (pkgoogle) [reply keras_hub/src/utils/timm/convert_efficientnet.py]: done

File: keras_hub/src/utils/timm/convert_efficientnet.py (hunk 14 above)
Comment (pkgoogle) [reply keras_hub/src/utils/timm/convert_efficientnet.py]: done

File: keras_hub/src/models/efficientnet/efficientnet_backbone.py (hunk 15 above)
Comment (pkgoogle) [reply keras_hub/src/models/efficientnet/efficientnet_backbone.py]: correct, done

**Comment by pkgoogle** [comment]
Remaining CI/CD failure appears unrelated to this PR.

**Review by mattdangerw — APPROVED** [review]
thanks!
